oauth2client/pkce.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2016 Google Inc. All rights reserved.
```

Comment:
As it happens, I'm working on an application that is both an OAuth2 client and an OAuth2 authorization server. :wink: 

We can now choose to either have an identical function in our own codebase, or import the 'private' function from `oauth2client`.